p2pserver/message/utils/msg_workerpool.go

```diff
@@ -0,0 +1,232 @@
+/*
+ * Copyright (C) 2018 The ontology Authors
+ * This file is part of The ontology library.
+ *
+ * The ontology is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * The ontology is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with The ontology.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package utils
+
+import (
+	"runtime"
+	"sync"
+	"time"
+
+	"github.com/ontio/ontology-eventbus/actor"
+	"github.com/ontio/ontology/common/log"
+	"github.com/ontio/ontology/p2pserver/message/types"
+	"github.com/ontio/ontology/p2pserver/net/protocol"
+)
+
+type msgJobChanList *[]*msgJobChan
+
+type msgWorkerPool struct {
+	maxWorkerCount    uint
+	maxWorkerIdleTime time.Duration
+	curWorkerCount    uint
+	stopChan          chan struct{}
+	waitingWokers     map[string]msgJobChanList
+	workChanPool      sync.Pool
+	lock              sync.Mutex
+}
+
+type msgJobItem struct {
+	msgPayload *types.MsgPayload
+	p2p        p2p.P2P
+	pid        *actor.PID
+	msgHandler MessageHandler
+}
+
+type msgJobChan struct {
+	lastUseTime time.Time
+	jobChan     chan *msgJobItem
+}
+
+func (this* msgWorkerPool) init() {
+
```

Comment:
remove these un-necessary blank lines?

Reply:
Yes, have removed

Reply:
adjust to `func NewWorkerPool() *msgWrokerPool`, constructing struct use two step is really ugly.

Reply:
Yes,  I have adjusted.